runtime/lib/component/custodian.js

```diff
@@ -1,235 +1,371 @@
-    return
-  }
-  return this.onNetworkDisconnect()
+  clearTimeout(this._clearTimeout)
+  this._bleTimeout = setTimeout(() => this._bluetoothStream.end(), 120 * 1000)
+  this._bluetoothStream.start(BLE_NAME, (err) => {
+    if (err) {
+      logger.error(err && err.stack)
+      logger.info('open ble failed, name', BLE_NAME)
+    } else {
+      logger.info('open ble success, name', BLE_NAME)
+      this.component.light.appSound('@yoda', 'system://wifi/setup_network.ogg')
+      this.component.light.play('@yoda', 'system://setStandby.js', {}, { shouldResume: true })
+    }
+    // FIXME(Yorkie): needs tell bind is unavailable?
+  })
 }
 
 // MARK: - Interception
-Custodian.prototype.turenDidWakeUp = function turenDidWakeUp () {
-  if (this.isPrepared()) {
+Custodian.prototype.turenDidWakeUp = function () {
+  if (this.isLoggedIn)
     return
-  }
+
   logger.warn('Network not connected, preparing to announce unavailability.')
   this.component.turen.pickup(false)
 
-  var currentAppId = this.component.lifetime.getCurrentAppId()
-  if (this.component.custodian.isConfiguringNetwork()) {
-    /**
-     * Configuring network, delegates event to network app.
-     */
-    logger.info('configuring network, renewing timer.')
-    return this.runtime.openUrl('yoda-skill://network/renew')
-  }
-
-  if (wifi.getNumOfHistory() === 0) {
-    if (currentAppId) {
-      /**
-       * although there is no WiFi history, yet some app is running out there,
-       * continuing currently app.
-       */
-      logger.info('no WiFi history exists, continuing currently running app.')
-      return this.component.light.ttsSound('@yoda', 'system://guide_config_network.ogg')
-        .then(() =>
-        /** awaken is not set for no network available, recover media directly */
-          this.component.turen.recoverPausedOnAwaken()
-        )
-    }
-    /**
-     * No WiFi connection history found, introduce device setup procedure.
-     */
-    logger.info('no WiFi history exists, announcing guide to network configuration.')
-    return this.component.light.ttsSound('@yoda', 'system://guide_config_network.ogg')
-      .then(() =>
-        /** awaken is not set for no network available, recover media directly */
-        this.component.turen.recoverPausedOnAwaken()
-      )
-  }
-
   /**
    * if runtime is logging in or network is unavailable,
    * and there is WiFi history existing,
    * announce WiFi is connecting.
    */
   logger.info('announcing network connecting on voice coming.')
-  wifi.enableScanPassively()
   return this.component.light.ttsSound('@yoda', 'system://wifi_is_connecting.ogg')
     .then(() =>
       /** awaken is not set for no network available, recover media directly */
       this.component.turen.recoverPausedOnAwaken()
     )
 }
-
 // MARK: - END Interception
+
+/**
+ * Set a flag which informs startup service that it is time to boot other services.
+ */
+Custodian.prototype.setStartupFlag = function () {
```

Comment:
Declearing this method at custodian component is weird a bit, custodian is designed to manage the network states, not the vui-daemon process state.